types/src/transaction/mod.rs

```diff
@@ -825,6 +825,18 @@ impl TransactionStatus {
             _ => Err(format_err!("Not Keep.")),
         }
     }
+
+    pub fn translate_invariant_violation(self, charge_invariant_violation: bool) -> Self {
```

Comment:
Is this the best solution? To go from invariant_violation -> discard -> keep anyway

Reply:
+1 this is janky